### PR TITLE
feat: check for new titanium cli version with update-notifier

### PR DESCRIPTION
**Pull request by caspahouzer**
Not every users is on slack nor reads the blog or just missed a cli update. Now, titanium prints out if there is an update

Bildschirmfoto 2022-05-25 um 19 52 28

### CODE REVIEW

File: lib/titanium.js [hunk 1]
```diff
@@ -24,6 +24,13 @@ var fs = require('fs'),
 	semver = require('semver'),
 	pkgJson = require('../package.json');
 
+// check if a new version of titanium CLI is available every hour and print an update notification
+const updateNotifier = require('update-notifier');
+const notifier = updateNotifier({ pkg: pkgJson, updateCheckInterval: 1000 * 60 * 60 });
+if (notifier.update && notifier.update.latest !== pkgJson.version) {
```
[the remark this answers is not in the document]
Comment (caspahouzer) [reply lib/titanium.js:30]: You have an example how to do that? I'm a real node newbie but try my best and will always learn

**Comment by m1ga** [comment]
still not convinced that we need that or at least with this package 😄 
The first issue in the package is `The dependency "boxen" throws security warnings on vulnerability scans.` which is nothing nice to add, the interval is very high and we don't update that often. If we update it is most of the time bundled with an SDK update so we don't even need to check it every build (+ 1 hour). 

As mentioned on Slack: perhaps integrate it into `ti info`. It collects all the CLI versions (and more) and we could check that agains a JSON file on the tidev servers. With that we could check java or node too and match that agains our recommended versions.

**Comment by cb1kenobi** [comment]
1. I would prefer the check happen async while the actual command runs, then display the results AFTER the command runs
2. We must disable the output when outputting JSON
3. I'm against saying [sudo]
4. I'm against adding this to the Alloy CLI because I strongly believe the Alloy CLI needs to die. Fold the Alloy CLI into the Titanium CLI/SDK
5. Lastly, I strongly prefer we use [check-kit v1.2.1](https://www.npmjs.com/package/check-kit) instead of update-notifier

**Comment by caspahouzer** [comment]
Can someone please overtake this PR? I got no time atm. If not, it's not a very critic PR and it's okay

**Pull request closed without merging**